### PR TITLE
Also error out when more than one argument is given on the command line.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

**Comment by tjhei** [comment]
I think we are going to be unable to pass PETSc command line arguments with this change (example ``./aspect test.prm -help``.

**Comment by bangerth** [comment]
This piece of code is executed after PETSc and MPI have already groked their arguments. It should only contain what's left for ASPECT to handle.

I was going to add `--help`/`-h` and `--version`/`-v` next. See also #1316.

**Comment by bangerth** [comment]
it should only contain -> `argc` and `argv` should only contain what's left for ASPECT to handle

**Comment by tjhei** [comment]
PETSc does not delete the args from argc/argv, so this patch would break the arguments. One workaround would be to put ``#ifndef ASPECT_USE_PETSC`` around this check.

**Comment by jedbrown** [comment]
The recommended approach is for the application to pull its arguments out of argv before passing the rest to PETSc.  If you wanted to PETSc to remove its arguments, how would it determine which options were meant for PETSc versus meant for the application?

**Comment by bangerth** [comment]
On 12/14/2016 10:58 AM, Jed Brown wrote:
> The recommended approach is for the application to pull its arguments out of
> argv before passing the rest to PETSc. If you wanted to PETSc to remove its
> arguments, how would it determine which options were meant for PETSc versus
> meant for the application?

Good point. Either way, that's not what we do right now. Will have to think 
about this some more then.

**Comment by bangerth** [comment]
OK, please take another look.

File: source/main.cc [hunk 1]
```diff
@@ -385,7 +385,22 @@ int main (int argc, char *argv[])
...
         print_aspect_header(std::cout);
 
-      if (argc < 2)
+      // check whether ASPECT was called with exactly one argument. if not,
+      // print an error message
+      //
+      // however, this does not work with PETSc because for PETSc, one
+      // may pass any number of flags on the command line; unfortunately,
+      // the PETSc initialization code (run through the call to
+      // MPI_InitFinalize above) does no filter these out, so all we
```
Comment (gassmoeller) [inline source/main.cc]: does *not*

**Pull request merged**